tools/create_sdk_cipd_unified_package.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# This script requires depot_tools to be on path.
+
+print_usage () {
+  echo "Usage: create_sdk_cipd_united_package.sh <PATH_TO_SDK_DIR> <VERSION_TAG>"
+  echo "  where:"
+  echo "    - PATH_TO_SDK_DIR is the path to the sdk folder"
+  echo "    - VERSION_TAG is the version of the package, e.g. 28r6 or 31v1"
+  echo ""
+  echo "This script downloads the packages specified in android_sdk_packages.txt and uploads"
+  echo "them to CIPD for linux, mac, and windows."
+  echo ""
+  echo "Manage the packages to download in 'android_sdk_packages.txt'. You can use"
+  echo "'sdkmanager --list --include_obsolete' in cmdline-tools to list all available packages."
+  echo "Packages should be listed in the format of <package-name>:<directory-to-upload>."
+  echo "For example, build-tools;31.0.0:build-tools"
+  echo "Multiple directories to upload can be specified by delimiting by additional ':'"
+  echo ""
+  echo "This script expects the cmdline-tools to be installed in your specified PATH_TO_SDK_DIR"
+  echo "and should only be run on linux or macos hosts."
+}
+
+# Validate version is provided
+if [[ $2 == "" ]]; then
+  print_usage
+  exit 1
+fi
+
+# Validate directory contains all SDK packages
+if [[ ! -d "$1" ]]; then
+  echo "Directory $1 not found."
+  print_usage
+  exit 1
+fi
+if [[ ! -d "$1/cmdline-tools" ]]; then
+  echo "SDK directory does not contain $1/cmdline-tools."
+  print_usage
+  exit 1
+fi
+
+platforms=("linux" "macosx" "windows")
+
+# Find the sdkmanager in cmdline-tools. We default to using latest if available.
+sdkmanager_path="$1/cmdline-tools/latest/bin/sdkmanager"
+find_results=()
+while IFS= read -r line; do
+  find_results+=("$line")
+done < <(find "$1/cmdline-tools" -name sdkmanager)
+i=0
+while [ ! -f "$sdkmanager_path" ]; do
+  if [ $i -ge ${#find_results[@]} ]; then
+    echo "Unable to find sdkmanager in the SDK directory. Please ensure cmdline-tools is installed."
+    exit 1
+  fi
+  sdkmanager_path="${find_results[$i]}"
```

Comment:
how to we know that this is the desired `sdkmanager`?  I think supporting the latest one from line 45 is probably sufficient

Reply:
I want this script to not explicitly only support latest. For example, currently, latest is equivalent to 5.0. It is valid to use the 5.0 version or just simply use latest. But, even though these are equivalent, sdkmanager stores them under different subdirs (`latest` and `5.0` respectively).

Also, when 6.0 is released, latest would point to 6.0 under the `latest` subdir while people who explicitly use the 5.0 package would still have the old package under the `5.0` subdir. I want this tool to not break for people who use explicit version numbers.

Reply:
got it 👍